test/utils.mjs

```diff
@@ -522,3 +522,9 @@ export const emptyL2 = async ({ nf3User, web3, logs }) => {
   }
   await new Promise(resolve => setTimeout(resolve, 6000));
 };
+
+export async function isTransactionMined(txHash, web3) {
+  const receipt = await web3.eth.getTransactionReceipt(txHash);
+  if (receipt !== null) return;
+  await isTransactionMined(txHash, web3);
```

Comment:
This function is recursive, but there isn't a limit for retrying, maybe one could add a timeout, even if it is a small one, just to avoid too many recursive calls and consequently prevent the heap from being out of memory.

Reply:
Agree. But not entirely sure about what to do.
When mentioning `setTimeOut`, do you mean inside the recursion? ie

```
// await getTransactionReceipt
// if condition for exiting
// else setTimeOut(waitTransactionToBeMined, 3000)
```

A counter might be possible too. Please let me know :pray:

Reply:
When I say a timeout I refer to a timeout from the perspective of the caller of the function `waitTransactionToBeMined` in in the case of the loop function it would be a counter.

Reply:
Okay: removed recursion, added timer and counter.